### ISSUE TITLE
PhoneGap Build discontinued

### ISSUE THREAD

**Issue by eskaigualker** [uncatcrea/wp-appkit#389]
How will we compile our applications from now on?
And more important, I'm afraid to ask this, but is the wp-appkit project still active?

Thank you

**Comment by mleroi** [comment]
Hi @eskaigualker,
Yes [the end of PhoneGap Build service](https://blog.phonegap.com/update-for-customers-using-phonegap-and-phonegap-build-cc701c77502c) is very bad news. Most probably the consequence for WP-AppKit will be that I will have to remove native exports and only keep Progressive Web App export.
As you pointed out, WP-AppKit is now a project that is not actively maintained, as we developers have moved towards new horizons. I try to keep the plugin working in its current version but probably no major features will be added in the future.
Best regards, and thank you for being one of our faithfull followers over the years.

**Comment by eskaigualker** [comment]
Yes it is bad news. Although these days, until your answer, I was being more optimistic. I've been testing alternatives like Monaca and Volt. And although I still haven't gotten them to compile correctly, I was hoping that a compatible export could be achieved from WP-Appkit. They don't look bad and I think they can even be cheaper than PGB.
Hopefully your project will continue, even slowly.
Anyway, thank you very much for your effort and your help.

**Comment by mleroi** [comment]
Thanks for the feedback, if you think that a WP-AppKit export for one of the alternative you are testing could be done without too much pain (simply export files in another directory structure or format for example) don't hesitate to tell me.

**Comment by eskaigualker** [comment]
Once again, thank you very much for your help.
To get to the structure that Voltbuider requires for, I have moved the  folders generated by Wp-appkit (core, lang, themes, vendor and index.html) to www and in index.html I have changed the path in line 21:

<script data-main="core/launch.js" src="vendor/require.js"></script>

It builds and apk, and I can even install it on a device, but doesn't work. I think (i hope)  it should not be hard to get it done.

thank you very much

![migrating](https://user-images.githubusercontent.com/1438079/94491879-5af5d480-01e9-11eb-9fab-8622ee60bd3d.jpg)
![volt](https://user-images.githubusercontent.com/1438079/94492263-35b59600-01ea-11eb-92d6-f581dc79188f.jpg)

**Comment by mleroi** [comment]
Great, I will test VoltBuilder too in the coming weeks. Tell me if you manage to make it work on your side!
For now I've focused on testing building WP-AppKit apps with raw Cordova, directly on my local computer, and I've managed to make it work quite well.
So I will probably add a "Cordova" export to WP-AppKit in the coming days. Cordova is not really straight forward to install and use but it allows users to build their apps on their own without requiring an external (and paid) service.
And then why not a VoltBuilder export also if we manage to get it working too.

**Comment by eskaigualker** [comment]
Well, it wasn't that difficult: with config.js in www folder, it works.
It was my first test, so it is too early to claim victory, but it looks like Voltbuilder is a good alternative for PGB. I have left moaca aside, because it seems more laborious (and expensive).
I think wp-appkit could be useful for a long time

**Comment by mleroi** [comment]
Hi @eskaigualker I just released a new version of WP-AppKit (1.6.0) that allows to export apps for Cordova and VoltBuilder.
You can take a look at the 2 corresponding tutorials : 
- [Build your WP-AppKit app with Cordova](https://uncategorized-creations.com/5153/build-your-wp-appkit-app-with-cordova/)
- [Build your WP-AppKit app with VoltBuilder](https://uncategorized-creations.com/5187/build-your-wp-appkit-app-with-voltbuilder/)

Don't hesitate for any feedback about those new exports, and again, many thanks for your tests and contributions.

**Comment by eskaigualker** [comment]
You leave me speechless. Thank you very much for your work and help.
